### PR TITLE
Improve Abha linking process

### LINKED ISSUE #6999: ABHA linking on patients
Todo:
 - [x] #7043
 - [x] `Already have an Abha number` should be there till the end of registration
 - [x] Add support to use Abha address for linking (check if this is directly possible)

**Describe the bug**
When a user initiates a ABHA generation process for a patient, only realising at the phase of entering "ABHA adress" that the patient already has a ABHA no. Then there is no way of going back to the form to link existing ABHA.

**To Reproduce**

1. Go to '.patient dashboard. https://care.coronasafe.in/facility/0c95c7f0-e1d2-4aff-83fa-933cef60d3a8/patient/7e8fc20f-02da-47f7-8c33-f2d8c7e52dd8/consultation/cb44a173-6fe2-452f-b2d1-6c6e2695411e
2. Click on '..Link ABHA.'
3. Enter Aadhar no, OTP, Mobile No. and mobile OTP
4. Finally the pop window will allow user to enter ABHA address. However, if the patient already has ABHA against the aadhar, an error at the bottom of the field says " Existing ABHA Address is used if ABHA Number already exists." CARE will not allow user to generate a new ABHA address. 
5. Now if the patient doesn't remember the ABHA address (but has access to the ABHA Number), there is nothing more user can do.
6. Even if the user closes the pop-up and again going to patient dashboard and clicks on "Link ABHA" to restart the process, this time to link existing ABHA, the same old pop-up with ABHA address input field appears.
Screenshot 2024-01-09 at 2 36 53 PM

Refreshing the patient dashboard and then taking the "Link ABHA" option fixes the problem. However user may not remember to refreash.

### PR DESCRIPTION
**Pull request by khavinshankar**
Resolves #6999 

Todo:
 - [ ] ~Have a draft for the Abha linking process, the user should be able to continue from where they left off before~ _(will be handled in a new pr)_
 - [x] `Already have an Abha number` should be there till the end of registration
 - [x] Add support to use Abha address for linking (check if this is directly possible)

### CODE REVIEW

File: src/Components/ABDM/LinkABHANumberModal.tsx [hunk 1]
```diff
@@ -159,11 +171,13 @@ const ScanABHAQRSection = ({
         value={qrValue}
         disabled={!!authMethods.length}
         onChange={(value) => {
-          if ([2, 7, 12].includes(value.length)) {
-            if (qrValue.length && qrValue[qrValue.length - 1] === "-") {
-              value.slice(value.length - 1);
-            } else {
-              value += "-";
+          if (value[0] && !isNaN(Number(value[0]))) {
+            if ([2, 7, 12].includes(value.length)) {
```
Comment (Ashesh3) [inline src/Components/ABDM/LinkABHANumberModal.tsx:176]: Can we add a small comment defining the purpose of these 3 magic numbers? `2, 7, 12`

**Review by Ashesh3 — APPROVED** [review]
LGTM

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
@khavinshankar 
image

**Comment by khavinshankar** [comment]
> @khavinshankar image

@nihal467 this is an existing issue, will create a new issue to handle this (it is a backend issue). It is caused because of trying to link the same Abha number multiple times

**Comment by nihal467** [comment]
LGTM

**Comment by nihal467** [comment]
LGTM

**Pull request merged**